front/src/common/Map/Buttons/MapButtons.js

```diff
@@ -12,28 +12,68 @@ import ButtonFullscreen from 'common/ButtonFullscreen';
 import MapSearch from 'common/Map/Search/MapSearch';
 import MapSettings from 'common/Map/Settings/MapSettings';
 import MapKey from 'common/Map/MapKey';
+import { useSelector, useDispatch } from 'react-redux';
+import { updateFeatureInfoClickOSRD } from 'reducers/osrdconf';
+
+import { getFeatureInfoClick } from 'reducers/osrdconf/selectors';
+import useOutsideClick from 'utils/hooks/useOutsideClick';
 import ButtonMapInfras from './ButtonMapInfras';
 
 export default function MapButtons(props) {
   const { resetPitchBearing, withInfraButton } = props;
-  const [showSearch, setShowSearch] = useState(false);
-  const [showSettings, setShowSettings] = useState(false);
-  const [showMapKey, setShowMapKey] = useState(false);
+  const featureInfoClick = useSelector(getFeatureInfoClick);
+  const dispatch = useDispatch();
+
+  const MAP_POPOVERS = { SEARCH: 'SEARCH', SETTINGS: 'SETTINGS', KEY: 'KEY' };
+
+  const [openedPopover, setOpenedPopover] = useState(undefined);
+
+  const toggleMapModal = (keyModal) => {
+    if (keyModal === openedPopover) {
+      keyModal = undefined;
+    }
+    setOpenedPopover(keyModal);
+
+    // Close the pop up of the map
+    if (featureInfoClick.displayPopup) {
+      dispatch(
+        updateFeatureInfoClickOSRD({
+          displayPopup: false,
+          feature: undefined,
+        })
+      );
+    }
+  };
```

Comment:
This is not useful anymore, you can remove it

Reply:
I need it to close the origin/destination menu when clicking on the settings menu.